include/lsst/afw/math/polynomials/Basis1d.h

```diff
@@ -0,0 +1,107 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2008-2018 LSST/AURA.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_AFW_MATH_POLYNOMIALS_Basis1d_h_INCLUDED
+#define LSST_AFW_MATH_POLYNOMIALS_Basis1d_h_INCLUDED
+#ifdef DOXYGEN
+
+namespace lsst { namespace afw { namespace math { namespace polynomials {
+
+/**
+ *  A basis concept for 1-d series expansions.
+ *
+ *  @note This class is only present in the documentation, as it represents an
+ *        abstract concept.
```

Comment:
I have never seen this done before. Clearly you could make this a real class by using template parameters. I assume you are trying to avoid inheritance, as per your documentation in polynomials.h ("the classes... are not polymorphic (no virtual functions)"). A few more words of explanation here would be appreciated.

Reply:
Concepts are very much a part of C++ - they're not in the language yet, but they've been a part of how people think about the language since the inception of the STL (see, e.g. http://en.cppreference.com/w/cpp/concept).  Sometimes people try to *implement* those with a templated base class (i.e. CRTP, as in the Eigen class hierarchies), and I tried that a bit here, but I ultimately rejected that approach because I felt the associated boilerplate obfuscated things more than it helped.

But this way of documenting a Concept as by telling Doxygen that it's a class is something new, here, and it seemed the best way to provide a separate documentation page that had more-or-less the right relationship to the concrete classes that implement the Concept.  I'm pretty happy with the result, but documentation is best judged by someone other than the author and I'm open to other ways of documenting this.  Given how focused it is on C++, it's a bit strange that Doxygen doesn't include something native for Concepts.

Reply:
A few words in the documentation would be helpful -- at least a link to the cppreference page you site. I had no idea the word "concept" was formal in this note.

Reply:
Good idea.  Will do.  I'll also capitalize Concept in the docs, which I should have done before.